src/script/op_codes.rs

```diff
@@ -36,92 +38,301 @@ pub fn to_boxed_hash(slice: &[u8]) -> Box<HashValue> {
+    CheckHeight(u64),
+    /// Pops the top of the stack as height and compares it to the current block height. Fails with INVALID_INPUT
+    /// if there is not a valid integer value on top of the stack. Fails with EMPTY_STACK if the stack is empty.
+    /// Fails with VERIFY_FAILED if the block height < height.
+    CompareHeightVerify,
+    /// Pops the top of the stack as height, then pushes the value of (height - the current height) to the stack. In
+    /// other words, this opcode replaces the top of the stack with the difference between that value and the current
+    /// height. Fails with INVALID_INPUT if there is not a valid integer value on top of the stack. Fails with
+    /// EMPTY_STACK if the stack is empty.
+    CompareHeight,
+
+    // Stack Manipulation
+    /// No op. Does nothing. Never fails.
+    Nop,
+    /// Pushes a zero onto the stack. This is a very common opcode and has the same effect as PushInt(0) but is more
+    /// compact. PushZero can also be interpreted as PushFalse (although no such opcode exists). Fails with
+    /// STACK_OVERFLOW if the stack would exceed the max stack height.
     PushZero,
-    /// Hash to top stack element with the Blake256 hash function and push the result to the stack
-    HashBlake256,
-    /// Fail the script immediately. (Must be executed.)
-    Return,
-    /// Drops the top stack item
+    /// Pushes a one onto the stack. This is a very common opcode and has the same effect as PushInt(1) but is more
+    /// compact. PushOne can also be interpreted as PushTrue, although no such opcode exists. Fails with STACK_OVERFLOW
+    /// if the stack would exceed the max stack height.
+    PushOne,
+    /// Push the associated 32-byte value onto the stack. Fails with INVALID_SCRIPT_DATA if HashValue is not a valid 32
+    /// byte sequence Fails with STACK_OVERFLOW if the stack would exceed the max stack height.
+    PushHash(Box<HashValue>),
+    /// Push the associated 64-bit signed integer onto the stack Fails with INVALID_SCRIPT_DATA if i64 is not a valid
+    /// integer. Fails with STACK_OVERFLOW if the stack would exceed the max stack height.
+    PushInt(i64),
+    /// Push the associated 32-byte value onto the stack. It will be interpreted as a public key or a commitment. Fails
+    /// with INVALID_SCRIPT_DATA if HashValue is not a valid 32 byte sequence Fails with STACK_OVERFLOW if the stack
+    /// would exceed the max stack height.
+    PushPubKey(Box<RistrettoPublicKey>),
+    /// Drops the top stack item. Fails with EMPTY_STACK if the stack is empty.
     Drop,
-    /// Duplicates the top stack item
+    /// Duplicates the top stack item. Fails with EMPTY_STACK if the stack is empty. Fails with STACK_OVERFLOW if the
+    /// stack would exceed the max stack height.
     Dup,
-    /// Reverse rotation. The top stack item moves into 3rd place, abc => bca
+    /// Reverse rotation. The top stack item moves into 3rd place, e.g. abc => bca. Fails with EMPTY_STACK if the stack
```

Comment:
```suggestion
    /// Reverse rotation. The top stack item moves into last place, e.g. abc => bca. Fails with EMPTY_STACK if the stack
```

Reply:
Not sure I agree with this one, and I did take the text directly from the RFC